src/main/java/sg/ncl/MainController.java

```diff
@@ -950,14 +953,14 @@ private void registerUserToDeter(JSONObject mainObject) throws
                     case USERNAME_ALREADY_EXISTS_EXCEPTION:
                         // throw from user service
                     {
-                        String email = mainObject.getJSONObject("user").getJSONObject("userDetails").getString("email");
+                        String email = mainObject.getJSONObject("user").getJSONObject("userDetails").getString(EMAIL);
                         log.warn("Register new users : email already exists: {}", email);
                         throw new UsernameAlreadyExistsException(ERROR_PREFIX + email + " already in use.");
                     }
                     case EMAIL_ALREADY_EXISTS_EXCEPTION:
                         // throw from adapter deterlab
                     {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Extract this nested code block into a method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1199)